### DISCUSSION TITLE
TensorFlow rebuild msg with AVX2 ? should I rebuild

### DISCUSSION THREAD

**Discussion by uzair004** [vladmandic/human#148] [Q&A] [answered]
While running my project tensorflow throws this message in command line.
` I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags`

I came to know that it is not warning rather a good thing, it will boost tensorflow performance if rebuild with AVX2.

_Should I rebuild ?_
_Will new build be compatible with running human version in my project ?_

Is there any easy way to rebuild ?

**Comment by vladmandic** [comment, marked as answer]
this means AVX2 is already enabled for performance-critical ops, just not enabled globally for all ops - so enabling it would result in very minimal difference.

and unfortunately, rebuilding `tensorflow` binaries and then rebuilding `tfjs-node` bindings to use new binaries is one of the most painful builds i've ever seen - definitely not worth it.

**Reply by uzair004** [reply]
Thanks
